doc/incubating-plugin.md

```diff
@@ -0,0 +1,273 @@
+
+# Using the incubating plugin
+
+The incubating plugin is in the `apollo-gradle-plugin-incubating`artifact. It is a rewrite of the plugin in Kotlin to make it more maintainable and have better support for multiple services.
+
+```
+buildscript {
+    // Replace
+    // classpath("com.apollographql.apollo:apollo-gradle-plugin")
+
+    // By
+    classpath("com.apollographql.apollo:apollo-gradle-plugin-incubating")
+}
+```
+
+# Differences with the current plugin
+
+### Plugin id is now `com.apollographql.apollo`
+
+We renamed the plugin id from `com.apollographql.android` to `com.apollographql.apollo` to make it clear that the plugin works also for non Android projects. `com.apollographql.android` will be removed in a future revision.
+
+```groovy
+// Instead of
+// apply plugin: 'com.apollographql.android'
```

Comment:
Actually, seeing the same thing in the `sourceSets` example below. Is it intentional?

Reply:
I was thinking that having them commented would convey the fact that they're not used anymore. But very happy to remove all theses.

Reply:
I think either makes sense. Commenting might make it obvious that they shouldn't be copy-pasted. If you keep them as comments, I'd double check the indents, because some of them are a bit messed up

Reply:
I removed the comments => No more messed up indents :) (hopefully)